src/visuals.js

```diff
@@ -294,69 +298,106 @@ export const Visuals = {
-      // this.p.fill(this.getNotGrey())
-      this.p.textSize(50)
-      this.p.textAlign(this.p.LEFT) // Right-align the text
-      this.p.text(`${this.frames} t`, 65, 50) // Adjust the x-coordinate to align the text
-      this.p.text(`${this.frameRate().toFixed(2)} fps`, 65, 100)
-    } else {
-      this.p.fill('white')
-      // this.p.rect(0, 0, 50, 20)
-      // this.p.fill('black')
-      // this.p.textAlign(this.p.RIGHT) // Right-align the text
-      const secondsAsTime = new Date(this.totalSec * 1000)
-        .toISOString()
-        .substr(14, 5)
-      const thisLevelSecondsAsTime = new Date(this.thisLevelSec * 1000)
-        .toISOString()
-        .substr(14, 5)
-      this.p.text('Total Frames: ' + this.preRun + this.frames, 50, 10) // Adjust the x-coordinate to align the text
-      this.p.text('Total Time: ' + secondsAsTime, 50, 20) // Adjust the x-coordinate to align the text
-      this.p.text('Total Shots: ' + this.missileCount, 50, 30) // Adjust the x-coordinate to align the text
-      this.p.text(
-        'Lvl ' +
-          (this.startingBodies - 2) +
-          ' - ' +
-          thisLevelSecondsAsTime +
-          ' - ' +
-          (this.startingBodies - this.bodies.length) +
-          '/' +
-          this.startingBodies +
-          ' - ' +
-          this.thisLevelMissileCount +
-          ' shots',
-        50,
-        40
-      ) // Adjust the x-coordinate to align the text
-      for (let i = 0; i < this.allLevelSec.length; i++) {
-        const prevLevel = this.allLevelSec[i]
-        const prevLevelSecondsAsTime = new Date(prevLevel.thisLevelSec * 1000)
-          .toISOString()
-          .substr(14, 5)
-        this.p.text(
-          'Lvl ' +
-            (this.allLevelSec.length - i) +
-            ' - ' +
-            prevLevelSecondsAsTime +
-            ' - ' +
-            prevLevel.thisLevelMissileCount +
-            ' shots',
-          50,
-          i * 10 + 50
-        ) // Adjust the x-coordinate to align the text
-      }
+    const { p } = this
+    p.push()
+    p.translate(20, 10)
+    p.fill('white')
+    p.noStroke()
+
+    const initialScoreSize = 60
+    this.scoreSize ||= initialScoreSize
+    p.textStyle(p.BOLDITALIC)
+    p.textAlign(p.LEFT, p.TOP)
+    const secondsLeft = this.timer / FPS
+
+    // when we have less than 10s left, flash the timer and make it huge
+    if (secondsLeft <= 0) {
+      p.noFill()
+      this.scoreSize = initialScoreSize
+    } else if (secondsLeft < 10 && this.scoreSize < 420) {
+      this.scoreSize += 5
+      p.fill(255, 255, 255, 150)
+    } else if (secondsLeft < 30 && this.scoreSize < 160) {
+      this.scoreSize += 2
+      p.fill(255, 255, 255, 150)
+    } else if (secondsLeft < 50 && this.scoreSize < 80) {
+      this.scoreSize += 1
+      p.fill(255, 255, 255, 150)
     }
+    p.textSize(this.scoreSize)
+
+    // TODO: remove this line when timer is ticked somewhere more suitable (okwme's branch)
+    this.timer--
```

Comment:
hmm think this will break current timer logic. Could merit changing existing logic but right now during initialization `this.timer` value is combined with `this.alreadyRun` so that when `this.frames >= this.timer` it means time as run out.

would changing back logic of line 71 and removing this line break anything else in your branch currently?

Reply:
i don't think it breaks anything as long as the timer ticks down somewhere as described in #82 

I first added this because timer was not moving, then changed that because the game was stopping at 30s (when timer and frames met)

Reply:
could we just use a new variable like `const timeLeft = this.timer - this.frames` ?

Reply:
sure. timer counts up in your branch?

Reply:
or like it is the frame at which the game ends?

Reply:
> or like it is the frame at which the game ends?
exactly

Reply:
ok got it